### DISCUSSION TITLE
希望有一个 broker 模块，实现子模块间的消息订阅与发布

### DISCUSSION THREAD

**Discussion by hevake** [cpp-main/cpp-tbox#13] [Ideas]
它应该有 invoke() 与 publish() 方法。
对应的有 register() 与 subscribe() 方法。

**Comment by guochaoyu** [comment]
基于topic id订阅消息似乎比topic string订阅效率高，
相同topic的消息subscribe端 收到消息顺序 应与publish端发出消息的顺序一致

**Comment by guochaoyu** [comment]
可以考虑消息缓存的配置：
 1 比如发送端缓存大小， 接收端缓存大小， 或者公用缓存大小
 2 某个topic独享或所有topic共享缓存等
 3 缓存满之后的处理规则，比如丢弃最新的消息还是丢弃最旧的消息

**Comment by baldwin-law** [comment]
类似mqtt ？ 好像已经有mqtt模块了。
